samples/va_intel/va_intel_interop.cpp

```diff
@@ -294,22 +329,50 @@ int main(int argc,char **argv)
     va_status = vaSyncSurface(va::display, surface_id);
     CHECK_VASTATUS(va_status, "vaSyncSurface");
 
-    dumpSurface(va::display, surface_id, argv[1]);
+    dumpSurface(va::display, surface_id, fn1, doInterop);
 
     cv::Size size(CLIP_WIDTH,CLIP_HEIGHT);
     cv::UMat u;
 
-    cv::vaapi::convertFromVASurface(surface_id, size, u);
+    t.start();
+    cv::va_intel::convertFromVASurface(va::display, surface_id, size, u);
     cv::blur(u, u, cv::Size(7, 7), cv::Point(-3, -3));
-    cv::vaapi::convertToVASurface(u, surface_id, size);
+    cv::va_intel::convertToVASurface(va::display, u, surface_id, size);
+    t.stop();
 
-    dumpSurface(va::display, surface_id, argv[2]);
+    dumpSurface(va::display, surface_id, fn2, doInterop);
 
     vaDestroySurfaces(va::display,&surface_id,1);
     vaDestroyConfig(va::display,config_id);
     vaDestroyContext(va::display,context_id);
 
-    vaTerminate(va::display);
+    return t.time(Timer::MSEC);
+}
+
+int main(int argc,char **argv)
+{
+    if (argc < 3)
+    {
+        fprintf(stderr,
+                "Usage: va_intel_interop file1 file2\n\n"
+                "where:  file1 is to be created, contains original surface data (NV12)\n"
+                "        file2 is to be created, contains processed surface data (NV12)\n");
+        exit(0);
+    }
+
+    if (!va::openDisplay())
+    {
+        fprintf(stderr, "Failed to open VA display for CL-VA interoperability\n");
+        exit(1);
+    }
+    fprintf(stderr, "VA display opened successfully\n");
+
+    float gpuTime = run(argv[1], argv[2], true);
+    float cpuTime = run(argv[1], argv[2], false);
```

Comment:
Is my understanding correct that second run does not create new OCL stuff (context device and queues) but uses the same created by first run of "run(true)"? If yes then second run uses old OCL stuff initialized for VA interoperability and it is not quite correct.